pkg/discoveryservice/manager.go

```diff
@@ -107,22 +107,33 @@ func (dsm *Manager) Start(ctx context.Context) {
 
 	// Start controllers
 	if err := (&envoycontroller.EnvoyConfigReconciler{
-		Client:   mgr.GetClient(),
-		Log:      ctrl.Log.WithName("controllers").WithName("envoyconfig"),
-		Scheme:   mgr.GetScheme(),
-		XdsCache: xdss.GetCache(EnvoyAPIV2),
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("envoyconfig"),
+		Scheme: mgr.GetScheme(),
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "envoyconfig")
 		os.Exit(1)
 	}
 
 	if err := (&envoycontroller.EnvoyConfigRevisionReconciler{
-		Client:   mgr.GetClient(),
-		Log:      ctrl.Log.WithName("controllers").WithName("envoyconfigrevision"),
-		Scheme:   mgr.GetScheme(),
-		XdsCache: xdss.GetCache(EnvoyAPIV2),
+		Client:     mgr.GetClient(),
+		Log:        ctrl.Log.WithName("controllers").WithName(fmt.Sprintf("envoyconfigrevision_%s", string(envoy.APIv2))),
+		Scheme:     mgr.GetScheme(),
+		XdsCache:   xdss.GetCache(envoy.APIv2),
+		APIVersion: envoy.APIv2,
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", fmt.Sprintf("envoyconfigrevision_%s", string(envoy.APIv2)))
+		os.Exit(1)
+	}
+
+	if err := (&envoycontroller.EnvoyConfigRevisionReconciler{
+		Client:     mgr.GetClient(),
+		Log:        ctrl.Log.WithName("controllers").WithName(fmt.Sprintf("envoyconfigrevision_%s", string(envoy.APIv3))),
+		Scheme:     mgr.GetScheme(),
+		XdsCache:   xdss.GetCache(envoy.APIv3),
+		APIVersion: envoy.APIv3,
```

Comment:
The 2 instances of the EnvoyConfigRevision controller, one for v2 and one for v3.